apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/account/service/AliasServiceImplTest.java

```diff
@@ -0,0 +1,341 @@
+package com.apollocurrency.aplwallet.apl.core.account.service;
+
+import com.apollocurrency.aplwallet.apl.core.account.dao.AliasOfferTable;
+import com.apollocurrency.aplwallet.apl.core.account.dao.AliasTable;
+import com.apollocurrency.aplwallet.apl.core.app.Alias;
+import com.apollocurrency.aplwallet.apl.core.app.AliasOffer;
+import com.apollocurrency.aplwallet.apl.core.app.Blockchain;
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.MessagingAliasAssignment;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.MessagingAliasSell;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author silaev-firstbridge on 3/19/2020
+ */
+@ExtendWith(MockitoExtension.class)
+class AliasServiceImplTest {
+    @Mock
+    private AliasTable aliasTable;
+    @Mock
+    private AliasOfferTable offerTable;
+    @Mock
+    private Blockchain blockchain;
+    @InjectMocks
+    @Spy
+    private AliasServiceImpl aliasService;
+
+    @Test
+    void shouldGetCount() {
+        //GIVEN
+
+        //WHEN
+        aliasService.getCount();
+
+        //THEN
+        verify(aliasTable, times(1)).getCount();
+    }
+
+    @Test
+    void shouldGetAccountAliasCount() {
+        //GIVEN
+        final long accountId = 7821792282123976600L;
+
+        //WHEN
+        aliasService.getAccountAliasCount(accountId);
+
+        //THEN
+        verify(aliasTable, times(1)).getCount(any(DbClause.LongClause.class));
+    }
+
+    @Test
+    void shouldGetAliasesByOwner() {
+        //GIVEN
+        final long accountId = 7821792282123976600L;
+        final int from = 1;
+        final int to = 2;
+
+        //WHEN
+        aliasService.getAliasesByOwner(accountId, from, to);
+
+        //THEN
+        verify(aliasTable, times(1)).getManyBy(any(DbClause.LongClause.class), eq(from), eq(to));
+    }
+
+    @Test
+    void shouldGetAliasByString() {
+        //GIVEN
+        final String aliasName = "AS1561618989348";
+
+        //WHEN
+        aliasService.getAlias(aliasName);
+
+        //THEN
+        verify(aliasTable, times(1)).getBy(any(DbClause.StringClause.class));
+    }
+
+    @Test
+    void shouldGetAliasesLike() {
```

Comment:
Maybe better to avoid impl details ('like' statement in database) on service level?

Reply:
done